miden-lib/asm/shared/util/account_id.masm

```diff
@@ -0,0 +1,387 @@
+#!
+#! Inputs:  [acct_id_prefix]
+#! Outputs: [is_non_fungible_faucet]
+#!
+#! Where:
+#! - acct_id_prefix is the prefix of the account ID.
+#! - is_non_fungible_faucet is a boolean indicating whether the account is a non-fungible faucet.
+export.is_non_fungible_faucet
+    exec.type push.NON_FUNGIBLE_FAUCET_ACCOUNT eq
+    # => [is_non_fungible_faucet]
+end
+
+#! Returns a boolean indicating whether the given account_ids are equal.
+#!
+#! Inputs:  [acct_id_prefix, acct_id_suffix, other_acct_id_prefix, other_acct_id_suffix]
+#! Outputs: [is_id_equal]
+#!
+#! Where:
+#! - acct_id_{prefix,suffix} are the prefix and suffix felts of an account ID.
+#! - other_acct_id_{prefix,suffix} are the prefix and suffix felts of the other account ID to compare against.
+#! - is_id_equal is a boolean indicating whether the account IDs are equal.
+export.is_id_eq
+    movup.2 eq
+    # => [is_prefix_equal, acct_id_suffix, other_acct_id_suffix]
+    movdn.2 eq
+    # => [is_suffix_equal, is_prefix_equal]
+    and
+    # => [is_id_equal]
+end
+
+#! Returns a boolean indicating whether the account is a faucet.
+#!
+#! Inputs:  [acct_id_prefix]
+#! Outputs: [is_faucet]
+#!
+#! Where:
+#! - acct_id_prefix is the prefix of the account ID.
+#! - is_faucet is a boolean indicating whether the account is a faucet.
+export.is_faucet
+    u32split drop push.FAUCET_ACCOUNT u32and eq.0 not
+    # => [is_faucet]
+end
+
+#! Returns a boolean indicating whether the account is a regular updatable account.
+#!
+#! Inputs:  [acct_id_prefix]
+#! Outputs: [is_updatable_account]
+#!
+#! Where:
+#! - acct_id_prefix is the prefix of the account ID.
+#! - is_updatable_account is a boolean indicating whether the account is a regular updatable
+#!   account.
+export.is_updatable_account
+    exec.type push.REGULAR_ACCOUNT_UPDATABLE_CODE eq
+    # => [is_updatable_account]
+end
+
+#! Returns a boolean indicating whether the account is a regular immutable account.
+#!
+#! Inputs:  [acct_id_prefix]
+#! Outputs: [is_immutable_account]
+#!
+#! Where:
+#! - acct_id_prefix is the prefix of the account ID.
+#! - is_immutable_account is a boolean indicating whether the account is a regular immutable
+#!   account.
+export.is_immutable_account
+    exec.type push.REGULAR_ACCOUNT_IMMUTABLE_CODE eq
+    # => [is_immutable_account]
+end
+
+#! Validates an account ID. Note that this does not validate anything about the account type,
+#! since any bit pattern is a valid account type.
+#!
+#! Inputs:  [account_id_prefix, account_id_suffix]
+#! Outputs: []
+#!
+#! Where:
+#! - account_id_{prefix,suffix} are the prefix and suffix felts of the account ID.
+#!
+#! Panics if:
+#! - account_id_prefix does not contain version zero.
+#! - account_id_prefix does not contain either the public or private storage mode.
+#! - account_id_suffix contains an anchor epoch that is greater or equal to 2^16.
+#! - account_id_suffix does not have its lower 8 bits set to zero.
+export.validate_id
```

Comment:
nit: we could probably rename this into just `validate` since it is already in the `account_id` module (i.e., it would probably be used as `exec.account_id::validate`.